src/protocols/multi_party_ecdsa/gg_2020/party_i.rs

```diff
@@ -286,13 +291,17 @@ impl Keys {
                         .correct_key_proof
                         .verify(&bc1_vec[i].e, zk_paillier::zkproofs::SALT_STRING)
                         .is_ok()
+                    && bc1_vec[i].e.n.bit_length() >= PAILLIER_MIN_BIT_LENGTH
+                    && bc1_vec[i].e.n.bit_length() <= PAILLIER_MAX_BIT_LENGTH
+                    && bc1_vec[i].dlog_statement.N.bit_length() >= PAILLIER_MIN_BIT_LENGTH
+                    && bc1_vec[i].dlog_statement.N.bit_length() <= PAILLIER_MAX_BIT_LENGTH
                     && bc1_vec[i]
                         .composite_dlog_proof_base_h1
-                        .verify(&bc1_vec[i].dlog_statement_base_h1)
+                        .verify(&bc1_vec[i].dlog_statement)
                         .is_ok()
                     && bc1_vec[i]
                         .composite_dlog_proof_base_h2
-                        .verify(&bc1_vec[i].dlog_statement_base_h2)
+                        .verify(&dlog_statement_base_h2)
```

Comment:
why not use directly `&bc1_vec[i].dlog_statement` ?

Reply:
`dlog_statement_base_h2` has a different order of arguments then `bc1_vec[i].dlog_statement`: `g` and `ni` are switched. I'm pretty sure that their order should be important for the `verify` method and indeed, when the change is made, all the tests fail.